htdocs/core/tpl/extrafields_view.tpl.php

```diff
@@ -243,7 +243,7 @@
 				if ($object->table_element == 'societe') {
 					$fieldid = 'socid';
 				}
-				print '<form enctype="multipart/form-data" action="'.$_SERVER["PHP_SELF"].'" method="post" name="formextra">';
+				print '<form enctype="multipart/form-data" action="'.$_SERVER["PHP_SELF"] . '?' . $fieldid . '=' . $object->id . '" method="post" name="formextra">';
```

Comment:
Because the method on form is POST, you must add the param as a hidden input field to avoid mixing parameters from url and post parameters.

Reply:
@eldy it is done 4 lines later : 
`print '<input type="hidden" name="'.$fieldid.'" value="'.$object->id.'">';`

Reply:
it's done like this for every other fields on card either directly or when calling $form->form_xxx : 
`		// Delivery date planed
		print '<tr><td>';
		$editenable = $usercancreate;
		print $form->editfieldkey("DateDeliveryPlanned", 'date_livraison', '', $object, $editenable);
		print '</td><td class="valuefield">';
		if ($action == 'editdate_livraison') {
			print '<form name="setdate_livraison" action="'.$_SERVER["PHP_SELF"].'?id='.$object->id.'" method="post">';
			print '<input type="hidden" name="token" value="'.newToken().'">';
			print '<input type="hidden" name="action" value="setdate_livraison">';
			print $form->selectDate($object->delivery_date ? $object->delivery_date : -1, 'liv_', 1, 1, '', "setdate_livraison", 1, 0);
			print '<input type="submit" class="button button-edit" value="'.$langs->trans('Modify').'">';
			print '</form>';
		} else {
			print $object->delivery_date ? dol_print_date($object->delivery_date, 'dayhour') : '&nbsp;';
			if ($object->hasDelay() && !empty($object->delivery_date)) {
				print ' '.img_picto($langs->trans("Late").' : '.$object->showDelay(), "warning");
			}
		}
		print '</td>';
		print '</tr>';

		// Delivery delay
		print '<tr class="fielddeliverydelay"><td>';
		$editenable = $usercancreate;
		print $form->editfieldkey("AvailabilityPeriod", 'availability', '', $object, $editenable);
		print '</td><td class="valuefield">';
		if ($action == 'editavailability') {
			$form->form_availability($_SERVER['PHP_SELF'].'?id='.$object->id, $object->availability_id, 'availability_id', 1);
		} else {
			$form->form_availability($_SERVER['PHP_SELF'].'?id='.$object->id, $object->availability_id, 'none', 1);
		}
		print '</td></tr>';
`